public/docs/_examples/cb-item-template/ts/app/simple-select.component.ts

```diff
@@ -0,0 +1,85 @@
+// #docplaster
+// #docregion
+import { Component } from '@angular/core';
+import { ContentChild } from '@angular/core';
+import { EventEmitter } from '@angular/core';
+import { Input } from '@angular/core';
+import { Output } from '@angular/core';
+import { TemplateRef } from '@angular/core';
+
+interface ItemContext {
+  item: any;
+  index: number;
+}
+
+// #docregion component
+// #docregion metadata
+@Component({
+  selector: 'simple-select',
+  template:
+  `
+    <button (click)='toggleItems()' class='select-root'>
+      
+      <template 
+        [ngTemplateOutlet]='itemTemplateRef'
+        [ngOutletContext]='{ item: value, index: -1 }'>
+      </template>
+
+    </button>
+  
+    <ul *ngIf='isShowingItems' class='select-items'>
+      <li *ngFor='let item of items ; let index = index ;' (click)='selectItem( item )'>
+        
+        <template 
+          [ngTemplateOutlet]='itemTemplateRef'
+          [ngOutletContext]='{ item: item, index: index }'>
+        </template>
+
+      </li>
+    </ul>
+  `
+})
+// #enddocregion metadata
+export class SimpleSelectComponent {
+
+  public isShowingItems: boolean;
+  @Input() public items: any[];
+  public itemTemplateRef: TemplateRef<ItemContext>;
+  @Input() public value: any;
+  @Output() public valueChange: EventEmitter<any>;
+
+  constructor() {
+    this.isShowingItems = false;
+    this.items = [];
+    this.itemTemplateRef = null;
+    this.value = null;
+    this.valueChange = new EventEmitter();
+  }
```

Comment:
I would side with @Foxandxss here: idiomatic ES6 and idiomatic TS are not the same and IMO there is no benefit in pretending that they are. IMO better to teach people idiomatic TS in TS examples (we all know that there is a lot of copy&paste going on)